pkg/controller/clusterinstallation/mattermost.go

```diff
@@ -223,75 +256,58 @@ func (r *ReconcileClusterInstallation) updateMattermostDeployment(mi *mattermost
 	// we will return and not upgrade the deployment.
 	if update {
 		reqLogger.Info(fmt.Sprintf("Running Mattermost image %s upgrade job check", imageName))
-
-		updateName := "mattermost-update-check"
-		job := &batchv1.Job{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      updateName,
-				Namespace: mi.GetNamespace(),
-			},
-			Spec: batchv1.JobSpec{
-				Template: corev1.PodTemplateSpec{
-					ObjectMeta: metav1.ObjectMeta{
-						Labels: map[string]string{"app": updateName},
-					},
-					Spec: *new.Spec.Template.Spec.DeepCopy(),
-				},
-			},
+		alreadyRunning, err := r.fetchRunningUpdateJob(mi, reqLogger)
+		if err != nil && k8sErrors.IsNotFound(err) {
+			reqLogger.Info("Launching update job")
+			if err = r.launchUpdateJob(mi, new, imageName, reqLogger); err != nil {
+				return errors.Wrap(err, "Launching update job failed")
+			}
+			return errors.New("Began update job")
```

Comment:
This shouldn't be an error, right?

Reply:
It should, actually, so that the reconcile loop remains in the Reconciling state. If we return nil, the loop may go back to Stable, which we don't want yet. Am I overlooking a better way to handle that state signalling?

Reply:
Alterntatively I could also leave a comment there for the sake of clarity since it is a little counterintuitive.

Reply:
Nope, actually looking at it again you've got it right 👍 I do think a comment to clarify would be useful to add

Reply:
Feel free to modify the comment in a future PR.